client/my-sites/plugins/minimum-jetpack-version-notice.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { warningNotice } from 'state/notices/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+import {
+	isJetpackSite,
+	siteHasMinimumJetpackVersion,
+	getSiteAdminUrl,
+	getSiteDomain
+} from 'state/sites/selectors';
+
+class MinimumJetpackVersionNotice extends Component {
+
+	static propTypes = {
+		translate: PropTypes.func.isRequired,
+		minimumJetpackVersionFailed: PropTypes.bool.isRequired,
+		adminUrl: PropTypes.string,
+		domain: PropTypes.string,
+	};
+
+	showWarning( { minimumJetpackVersionFailed, domain, adminUrl, triggerNotice, translate, siteId } ) {
+		if ( minimumJetpackVersionFailed ) {
+			triggerNotice(
+				translate(
+					'Jetpack %(version)s is required to take full advantage of plugin management in %(site)s.',
+					{
+						args: {
+							version: config( 'jetpack_min_version' ),
+							site: domain
+						}
+					}
+				), {
+					button: translate( 'Update now' ),
+					href: adminUrl,
+					id: 'allSitesNotOnMinJetpackVersion' + config( 'jetpack_min_version' ) + '-' + siteId,
+					displayOnNextPage: true
+				}
+			);
+		}
+	}
+
+	componentWillMount() {
+		this.showWarning( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		this.showWarning( nextProps );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect(
+	state => {
+		const selectedSiteId = getSelectedSiteId( state );
+		return {
+			siteId: selectedSiteId,
+			adminUrl: getSiteAdminUrl( state, selectedSiteId, 'plugins.php?plugin_status=upgrade' ),
+			domain: getSiteDomain( state, selectedSiteId ),
+			minimumJetpackVersionFailed: !! isJetpackSite( state, selectedSiteId ) &&
```

Comment:
Hi @tyxla, We want to return true or false the first selector may return null, and in that case as the propTypes say the Boolean is required we would not be respecting it. The !! forces a true or false value as required in our proptypes.